### PR TITLE
Topic detailview enhancments

### PR DESCRIPTION
**Pull request by CHenrie**

### CODE REVIEW

**Comment by CHenrie** [comment]
@jbest84  Please start review I know there need to be changes ans Doc Comments

File: src/Application.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ define('Mobile/SalesLogix/Application', [
     'Sage/Platform/Mobile/Application',
     'dojo/sniff',
     'moment'
+
```
Comment (jbest84) [inline src/Application.js]: Remove blank line

File: src/Application.js [hunk 2]
```diff
@@ -104,7 +105,7 @@ define('Mobile/SalesLogix/Application', [
         homeViewId: 'myactivity_list',
         loginViewId: 'login',
         logOffViewId: 'logoff',
-
+        ModelManager: null,
```
Comment (jbest84) [inline src/Application.js]: I can't find any usages of this within src

File: src/Application.js [hunk 3]
```diff
@@ -123,6 +124,7 @@ define('Mobile/SalesLogix/Application', [
                 request.setRequestHeader('X-Application-Version', string.substitute('${major}.${minor}.${revision}', app.mobileVersion));
                 original.apply(this, arguments);
             };
+
```
Comment (jbest84) [inline src/Application.js]: Remove

File: src/Application.js [hunk 4]
```diff
@@ -529,9 +531,14 @@ define('Mobile/SalesLogix/Application', [
             this.requestUserOptions();
             this.requestSystemOptions();
             this.setDefaultMetricPreferences();
+            this.initQuickFormServices();
         },
         onRequestUserDetailsFailure: function(response, o) {
         },
+        initQuickFormServices: function () {
+            var qfs = App.serviceManager.get('quickFormService');
```
Comment (jbest84) [inline src/Application.js]: No need to use "App" here, App is equal to "this".

File: src/Application.js [hunk 5]
```diff
@@ -529,9 +531,14 @@ define('Mobile/SalesLogix/Application', [
...
+            qfs.initModelData();
```
Comment (jbest84) [inline src/Application.js]: We should probably check this variable for null/undefined before calling methods on it

File: src/Models/EntityModel.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2014, SalesLogix, Inc. All rights reserved.
```
Comment (jbest84) [inline src/Models/EntityModel.js]: Wrong copyright

File: src/Models/EntityModel.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+         * @property {String}
+         * The unique (within the current form) name of the model
+         */
+        name: 'EntityModel',
+        dispalyName: 'EntityModel',
+        displayNamePlural: 'Entity Models',
+        modelData: {},
+        entityType: null,
+        service: null,
+        contractName:'Dynamic',
```
Comment (jbest84) [inline src/Models/EntityModel.js]: should be lower case

File: src/Models/QuickFormControls/AddressControl.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/AddressControl.js]: Wrong copyright

File: src/Models/QuickFormControls/BooleanControl.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/BooleanControl.js]: Wrong copyright

File: src/Models/QuickFormControls/CurrencyControl.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/CurrencyControl.js]: Wrong copyright

File: src/Models/QuickFormControls/DateControl.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/DateControl.js]: Wrong copyright

File: src/Models/QuickFormControls/EmailControl.js [hunk 12]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/EmailControl.js]: Wrong copyright

File: src/Models/QuickFormControls/LookupControl.js [hunk 13]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/LookupControl.js]: Wrong copyright

File: src/Models/QuickFormControls/NameControl.js [hunk 14]
```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/NameControl.js]: Wrong copyright

File: src/Models/QuickFormControls/NumericControl.js [hunk 15]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/NumericControl.js]: Wrong copyright

File: src/Models/QuickFormControls/OwnerControl.js [hunk 16]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/OwnerControl.js]: Wrong copyright

File: src/Models/QuickFormControls/PhoneControl.js [hunk 17]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/PhoneControl.js]: Wrong copyright

File: src/Models/QuickFormControls/PicklistControl.js [hunk 18]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/PicklistControl.js]: Wrong copyright

File: src/Models/QuickFormControls/TextControl.js [hunk 19]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/TextControl.js]: Wrong copyright

File: src/Models/QuickFormControls/UrlControl.js [hunk 20]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/UrlControl.js]: Wrong copyright

File: src/Models/QuickFormControls/UserControl.js [hunk 21]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/UserControl.js]: Wrong copyright

File: src/Models/QuickFormControls/_BaseControl.js [hunk 22]
```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 1997-2013, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormControls/_BaseControl.js]: Wrong copyright

File: src/Models/QuickFormModel.js [hunk 23]
```diff
@@ -0,0 +1,184 @@
+/* Copyright (c) 2014, SalesLogix, Inc. All rights reserved.
```
Comment (jbest84) [inline src/Models/QuickFormModel.js]: Wrong copyright

File: src/Models/QuickFormModel.js [hunk 24]
```diff
@@ -0,0 +1,184 @@
...
+        dispalyName: 'Quick Form Model',
+        displayNamePlural: 'Quick Form Models',
+        modelData: {},
+        entityType: null,
+        resourceKind: null,
+        select: [],
+        include: [],
+        layout: [],
+        controlMaps:[
+            { name: 'text', type: 'Sage.Platform.QuickForms.Controls.QFTextBox, Sage.Platform.QuickForms', ctor: TextControl},
```
Comment (jbest84) [inline src/Models/QuickFormModel.js]: Is there any reason we can't register the controls like we do with our fieldmanager? Then we could move the "type" to the control itself, and iterate the control collection for that type/name.

File: src/Services/EntityService.js [hunk 25]
```diff
@@ -0,0 +1,102 @@
+/* Copyright (c) 2014, SalesLogix, Inc. All rights reserved.
```
Comment (jbest84) [inline src/Services/EntityService.js]: Wrong copyright

File: src/Services/QuickFormService.js [hunk 26]
```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2014, SalesLogix, Inc. All rights reserved.
```
Comment (jbest84) [inline src/Services/QuickFormService.js]: Wrong copyright

File: src/Services/QuickFormService.js [hunk 27]
```diff
@@ -0,0 +1,75 @@
...
+        initModelData: function () {
+            var dataPromise, queryOptions;
+            queryOptions = {
+                where: "DefinitionType eq 'Sage.Platform.QuickForms.MobileQuickFormMainDetailViewDefinition, Sage.Platform.QuickForms'"
+            };
+            
+            dataPromise = this.getModels(queryOptions);
+            dataPromise.then(function (models) {
+                if (models) {
+
```
Comment (jbest84) [inline src/Services/QuickFormService.js]: What is going on here?

File: src/Services/QuickFormService.js [hunk 28]
```diff
@@ -0,0 +1,75 @@
...
+                }
+            });
+
+        },
+
```
Comment (jbest84) [inline src/Services/QuickFormService.js:74]: Remove blank lines

File: src/Services/_ModelServiceBase.js [hunk 29]
```diff
@@ -0,0 +1,162 @@
+/* Copyright (c) 2014, SalesLogix, Inc. All rights reserved.
```
Comment (jbest84) [inline src/Services/_ModelServiceBase.js]: Wrong copyright

File: src/Services/_ModelServiceBase.js [hunk 30]
```diff
@@ -0,0 +1,162 @@
...
+            lang.mixin(this, o);
+        },
+        init: function () {
+            //this.initModelData();
+        },
+        initModelData: function () {
+            var dataPromise;
+            dataPromise = this.getModels();
+            dataPromise.then(function (models) {
+                if (models) {
```
Comment (jbest84) [inline src/Services/_ModelServiceBase.js:106]: Empty if

File: src/Services/_ModelServiceBase.js [hunk 31]
```diff
@@ -0,0 +1,162 @@
...
+                     .setQueryArg('_indented', 'true');
+            return request;
+
+        },
+        getModelsRequest: function (queryOptions) {
+            var request;
+            request = new Sage.SData.Client.SDataResourceCollectionRequest(this.service);
+            request.setResourceKind(this.resourceKind);
+            request.setContractName(this.contractName);
+            request.setQueryArg('_indented', 'true');
```
Comment (jbest84) [inline src/Services/_ModelServiceBase.js]: We might want to set this as a dev only option

File: src/Views/Account/Detail.js [hunk 32]
```diff
@@ -165,10 +171,10 @@ define('Mobile/SalesLogix/Views/Account/Detail', [
                             iconClass: 'fa fa-edit fa-lg',
                             action: 'addNote'
                         }]
-                }, {
+            },{
                     title: this.detailsText,
                     name: 'DetailsSection',
-                    children: [{
+                      children: [{
```
Comment (jbest84) [inline src/Views/Account/Detail.js]: Extra whitespace

File: src/Views/Account/Detail.js [hunk 33]
```diff
@@ -276,7 +282,8 @@ define('Mobile/SalesLogix/Views/Account/Detail', [
                             title:  this.relatedAttachmentTitleText
                         }]
                 }]);
-        }
+        },
```
Comment (jbest84) [inline src/Views/Account/Detail.js]: Trailing comma

File: src/Views/Lead/Detail.js [hunk 34]
```diff
@@ -13,15 +13,18 @@ define('Mobile/SalesLogix/Views/Lead/Detail', [
     'dojo/_base/declare',
     'dojo/string',
     'Mobile/SalesLogix/Format',
-    'Sage/Platform/Mobile/Detail'
+    'Sage/Platform/Mobile/Detail',
+    '../_QuickFormDetailMixin',
+
```
Comment (jbest84) [inline src/Views/Lead/Detail.js]: Remove blank line

File: src/Views/QuickFormDetailWidget.js [hunk 35]
```diff
@@ -0,0 +1,142 @@
+/* Copyright (c) 2010, Sage Software, Inc. All rights reserved.
```
Comment (jbest84) [inline src/Views/QuickFormDetailWidget.js]: Wrong copyright

File: src/Views/QuickFormDetailWidget.js [hunk 36]
```diff
@@ -0,0 +1,142 @@
...
+    'dojo/dom-attr',
+    'dojo/_base/connect',
+    'dojo/_base/array',
+    'Sage/Platform/Mobile/Utility',
+    'Sage/Platform/Mobile/Format',
+    'dijit/_Widget',
+    'Sage/Platform/Mobile/_Templated',
+    'Sage/Platform/Mobile/RelatedViewManager',
+    'Sage/Platform/Mobile/RelatedViewDetailWidget'
+
```
Comment (jbest84) [inline src/Views/QuickFormDetailWidget.js]: Remove blank line

File: src/Views/_QuickFormDetailMixin.js [hunk 37]
```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 1997-2015, SalesLogix, NA., LLC. All rights reserved.
```
Comment (jbest84) [inline src/Views/_QuickFormDetailMixin.js]: Wrong copyright

File: src/Views/_QuickFormDetailMixin.js [hunk 38]
```diff
@@ -0,0 +1,74 @@
...
+    declare,
+    array,
+    lang,
+    aspect
+   
+) {
+    return declare('Mobile.SalesLogix.Views._QuickFormDetailMixin', null, {
+       
+        entityName: null,
+        o_createLayout: null,
```
Comment (jbest84) [inline src/Views/_QuickFormDetailMixin.js]: rename

File: src/Views/_QuickFormDetailMixin.js [hunk 39]
```diff
@@ -0,0 +1,74 @@
...
+            quickFormSection = {
+                title: '',
+                list: true,
+                name: 'QuickFormViews',
+                children: [{
+                    name: this.entityName + '_quickFormDetail',
+                    relatedView: {
+                        widgetType: 'quickFormDetail',
+                        id: this.entityName + '_quickFormDetail'
+                        //quickFormName:  this.entityName + 'MobileDetail'
```
Comment (jbest84) [inline src/Views/_QuickFormDetailMixin.js]: Remove commented out code

**Pull request closed without merging**